src/Functions/TargetSpecific.cpp

```diff
@@ -0,0 +1,64 @@
+#include <Functions/TargetSpecific.h>
+
+#if defined(__GNUC__)
+#   include <cpuid.h>
+#   include <x86intrin.h>
+#else
+#   error "Only CLANG and GCC compilers are supported for dynamic dispatch"
+#endif
+
+namespace DB
+{
+
+__attribute__ ((target("xsave")))
+UInt64 xgetbv(UInt32 ecx)
+{
+    return _xgetbv(ecx);
+}
+
+UInt32 GetSupportedArches()
```

Comment:
BTW, it's already third implementation of CPU features detection :)
Previous:
- `libcpuid` from `contrib`;
- `Common/CpuId.h`